cin_validator/rules/cin_2023_24/rule_1530.py

```diff
@@ -0,0 +1,136 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import (
+    CINTable,
+    IssueLocator,
+    RuleContext,
+    rule_definition,
+)
+from cin_validator.test_engine import run_rule
+
+ChildIdentifiers = CINTable.ChildIdentifiers
+UPN = ChildIdentifiers.UPN
+
+
+@rule_definition(
+    code=1530,
+    module=CINTable.ChildIdentifiers,
+    message="UPN invalid (characters 2-4 not a recognised LA code)",
+    affected_fields=[UPN],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    df = data_container[ChildIdentifiers]
+    """
+    If <UPN> (N00001) present then characters 2-4 of <UPN> must be a valid post April 1998 LA code 
+    or a recognised ‘pseudo LA’ code 
+
+    001-005, 201-213, 301-320, 330-336, 340-344, 350-359, 370-373, 380-384, 390-394, 420, 660-681, 
+    701-708, 800-803, 805-808, 810-813, 815, 816, 820- 823, 825, 826, 830, 831, 835-837, 838-839, 
+    840, 841, 845, 846, 850-852, 855-857, 860, 861, 865-896, 908, 909, 916, 919, 921, 925, 
+    926, 928, 929, 931, 933, 935-938, 940-941
+    """
+    LA_list = []
+    LA_list.extend("00" + str(x) for x in range(1, 6))
+    LA_list.extend(range(201, 214))
+    LA_list.extend(range(301, 321))
+    LA_list.extend(range(330, 337))
+    LA_list.extend(range(340, 345))
+    LA_list.extend(range(350, 360))
+    LA_list.extend(range(370, 374))
+    LA_list.extend(range(380, 385))
+    LA_list.extend(range(390, 395))
+    LA_list.extend(range(660, 682))
+    LA_list.extend(range(701, 709))
+    LA_list.extend(range(800, 804))
+    LA_list.extend(range(805, 809))
+    LA_list.extend(range(810, 814))
+    LA_list.extend(range(820, 824))
+    LA_list.extend(range(835, 840))
+    LA_list.extend(range(850, 853))
+    LA_list.extend(range(855, 858))
+    LA_list.extend(range(865, 897))
+    LA_list.extend(range(935, 939))
+    LA_list.extend(range(940, 942))
```

Comment:
Side note:
Extending the upper bound of this range object to 944 would've added 943 and 942 too.